src/Relude/Extra/Enum.hs

```diff
@@ -23,6 +26,27 @@ import Relude
 universe :: (Bounded a, Enum a) => [a]
 universe = [minBound .. maxBound]
 
+{- | Create a function that parses values of type @a@
```

Comment:
This function doesn't create function that parses values. It creates function that is an inverse of given function. You can look at `inverseMap` as at function that converts function of type `(a -> k)` to function of type `(k -> Maybe a)`.